src/Parser/Mage/Frost/Modules/Features/FrostBomb.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import { formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import AbilityTracker from 'Parser/Core/Modules/AbilityTracker';
+import Module from 'Parser/Core/Module';
+
+// Frost Bomb cleaves don't always hit on identical timestamps, so we're giving a 100ms buffer
+const PROC_WINDOW_MS = 100;
+
+class FrostBomb extends Module {
+  static dependencies = {
+    combatants: Combatants,
+    abilityTracker: AbilityTracker,
+	}
+
+  damage = 0;
+  hits = 0;
+  procs = 0;
+  hitTimestamp;
+
+  on_initialized() {
+	   this.active = this.combatants.selected.hasTalent(SPELLS.FROST_BOMB_TALENT.id);
+  }
+
+  on_byPlayer_damage(event) {
+    if(event.ability.guid === SPELLS.FROST_BOMB_DAMAGE.id) {
```

Comment:
The nesting isn't awful here but if this were to get worse you could bomb out by checking for a non-equivalency and returning.